### DISCUSSION TITLE
Adding cmp-git source to cmp

### DISCUSSION THREAD

**Discussion by Eryx5502** [LazyVim/LazyVim#2104] [Q&A] [answered]
I am trying to get the 'git' source for nvim-cmp working on my lazyvim and tried the snippet in the receipes of LazyVim's web
```lua
  {
    "hrsh7th/nvim-cmp",
    dependencies = {
      "petertriho/cmp-git",
      dependencies = "nvim-lua/plenary.nvim",
    },
    ---@param opts cmp.ConfigSchema
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.sources = cmp.config.sources(vim.list_extend(opts.sources, { { name = "git" } }))
    end,
  },
```

But this is not working and I'm a bit lost about how to get it working. It did installed the plugin when I restarted nvim, but the completion does not work and `:CmpStatus` says the 'git' source is unknown.

Am I doing something wrong or does this look like a problem with cmp or cmp-git and thus I should ask in their repos?

**Comment by dpetka2001** [comment]
https://github.com/petertriho/cmp-git/issues/53

**Reply by Eryx5502** [reply]
I guess that's it, thanks. Will subscribe to that thread to get updated when it's solved and then I'll try again.

**Reply by dpetka2001** [reply]
For me it doesn't work in `Octo` buffers only. How are you using it?

**Reply by Eryx5502** [reply]
I would mainly use it for octo myself, but it also tried and it doesn't work on the commit descriptions when using nvim as editor.

**Reply by dpetka2001** [reply, marked as answer]
In Octo you can press `<C-x><C-o>` for its internal omnifunc autocompletion it provides and it will work correctly as far as I know.
I have the following spec for `cmp-git`
```lua
  {
    "nvim-cmp",
    dependencies = {
      {
        "petertriho/cmp-git",
        opts = {
          -- remotes = { "upstream" },
        },
      },
    },
    opts = function(_, opts)
      table.insert(opts.sources, #opts.sources + 1, {
        name = "git",
        priority = 500,
        group_index = 1,
      })
    end,
  },
```
and it works correctly when I do `git commit` and Neovim gets invoked as default editor.

**Reply by Eryx5502** [reply]
Alright, I'll try and see if it works that way for me. I knew about octo omnifunc but didn't work either for me. Silly question, do I have to release ctrl between <c-x> and <c-o> presses?

**Reply by dpetka2001** [reply]
Nope.

**Reply by Eryx5502** [reply]
That's what I thought. I will try again just in case and also try with the imaps on Octo's docs. If it doesn't work finally I guess I should also open an issue for octo.

**Reply by Eryx5502** [reply]
Works with your config everywhere except in octo! Let's wait for the [issue](https://github.com/petertriho/cmp-git/issues/53) in cmp-git to get solved.

**Reply by dpetka2001** [reply]
Yeah in `octo` I use `<c-x><c-o>`. Works fine for me.
